### ISSUE TITLE
Add HeroDetails Screen

### ISSUE THREAD

**Issue by ilkeka** [ilkeraslan/dota2compose#1]
We should have a `HeroDetails` Screen which is capable of showing a Hero's details.

Specifically this screen should have:
- [ ] Hero's image
- [ ] Hero's name and type
- [ ] Some of the main stats of a Hero (like `baseHealth`, `moveSpeed`, `baseMana` etc.)

The Screen should be built using Jetpack Compose.

**Comment by poseidon2060** [comment]
Can I work on this issue?
I am a bit new to compose and android dev, I hope I can solve this issue and learn along the way.

**Comment by ilkeraslan** [comment]
> Can I work on this issue? I am a bit new to compose and android dev, I hope I can solve this issue and learn along the way.

Of course, I am assigning this to you. Let me know if you have any difficulties so that we can discuss.

**Comment by ilkeraslan** [comment]
> also I noticed that when I ran the code, it was giving the following error:
> 
> `Class content provided for type descriptor me.ilker.dota2compose.ui.teams.TeamsKt$teamDetails$1$1$1$1 actually defines class me.ilker.dota2compose.ui.teams.TeamsKt$TeamDetails$1$1$1$1`
> 
> there seems to be an error in `Heroes.kt` file where it calls `R.drawable.ic_error` but I cannot point out what the error is

I am now aware of this error, but if you want I can create a separate issue for this and assign it to you if you want to investigate further. Let me know!

**Comment by poseidon2060** [comment]
> > Can I work on this issue? I am a bit new to compose and android dev, I hope I can solve this issue and learn along the way.
> 
> Of course, I am assigning this to you. Let me know if you have any difficulties so that we can discuss.

Thankyou, I'll try my best and will ask any doubts that I have.
Is this the channel to ask doubts/discuss or is there some discord server?

**Comment by poseidon2060** [comment]
Hey @ilkeraslan ! let me know if I got this right
I should implement a bottom sheet that is opened up when I tap on a specific hero in `Heroes` screen and that bottom sheet should show Details of heroes as mentioned above in the checklist

Can you guide me slightly on how to proceed with this issue? What I mean by that is what all things I need to implement to achieve the above goal and how to start on it. That would be a huge help, meanwhile I'll try to understand the code and experiment implementing it myself

**Comment by ilkeraslan** [comment]
> Hey @ilkeraslan ! let me know if I got this right I should implement a bottom sheet that is opened up when I tap on a specific hero in `Heroes` screen and that bottom sheet should show Details of heroes as mentioned above in the checklist?
> 
> Can you guide me slightly on how to proceed with this issue? What I mean by that is what all things I need to implement to achieve the above goal and how to start on it. That would be a huge help, meanwhile I'll try to understand the code and experiment implementing it myself

Hey @poseidon2060, for the sake of simplicity I would only implement the `HeroDetails` composable which includes the given details that I have described in the issue. This issue should resolve only how the details are shown, so do not worry about the navigation ecc. If you want you can work with the preview without connecting the screen.

Let me know if this helps you, and graphically do not worry too much if it's perfect or not. What would be great is to see the image of a hero, some of its details like we would see in a normal details screen. Feel free to think the UI as you want, lets see together what comes out of this :smile:

**Comment by poseidon2060** [comment]
Thankyou @ilkeraslan got it, working on it😄💪
This composable lives in the `me/ilker/dota2compose/ui/screens` path right?

**Comment by ilkeraslan** [comment]
> Thankyou @ilkeraslan got it, working on it😄💪 This composable lives in the `me/ilker/dota2compose/ui/screens` path right?

I guess so, yes 👍
